google/stackdriver_monitoring/server_handlers.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implements commands that starts the web-server daemon."""
+
+
+import logging
+import threading
+import time
+import traceback
+from http_server import HttpServer
+
+import command_processor
+import spectator_client
+import stackdriver_service
+import datadog_service
+
+
+class HomePageHandler(command_processor.CommandHandler):
+  """Implements the home page for the server.
+
+  This lists all the commands with links to execute them.
+  """
+  def __init__(self, all_handlers, url_path, command_name, description):
+    """Constructor.
+
+    Args:
+      all_handlers: [list of CommandHandler] Determines the page contents.
+        The entries that have url_paths will be displayed.
+    """
+    super(HomePageHandler, self).__init__(url_path, command_name, description)
+    self.__all_handlers = all_handlers
+
+  def process_web_request(self, request, path, params, fragment):
```

Comment:
This is for the home page. I dont care. The command is intended to support a web browser in HTML. I'm not going to support an alternative. The reason I check other commands for HTML is that I'm reusing the same logic for command line and web invocation and only want HTML for web. I've been moving away from that (having split process_web_request vs process_commandline_request). This particular handler doesnt give itself a command name, so isnt even visible or invocable from the commandline (using the standard dispatch mechanism).